src/main/java/uk/gov/ch/developer/docs/controller/developer/SignOutController.java

```diff
@@ -1,57 +1,44 @@
 package uk.gov.ch.developer.docs.controller.developer;
 
 import java.io.IOException;
-import java.util.Map;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import uk.gov.ch.developer.docs.DocsWebApplication;
+import uk.gov.ch.oauth.Oauth2;
 import uk.gov.ch.oauth.identity.IIdentityProvider;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
 import uk.gov.companieshouse.session.Session;
-import uk.gov.companieshouse.session.SessionKeys;
 import uk.gov.companieshouse.session.handler.SessionHandler;
 import uk.gov.companieshouse.session.store.Store;
 
 @Controller
 public class SignOutController {
     private static final Logger LOGGER =
             LoggerFactory.getLogger(DocsWebApplication.APPLICATION_NAME_SPACE);
-    private static final String SIGN_IN_INFO = SessionKeys.SIGN_IN_INFO.getKey();
 
     @Autowired
     private IIdentityProvider identityProviders;
     @Autowired
     private Store store;
+    @Autowired
+    private Oauth2 oauth2;
+
+    public SignOutController() {
+    }
 
     @GetMapping("${signout.url}")
     public void doSignOut(final HttpServletResponse httpServletResponse,
-            final HttpServletRequest httpServletRequest) throws IOException {
+                          final HttpServletRequest httpServletRequest) throws IOException {
 
         LOGGER.debug("Processing sign out");
 
         final Session chSession = (Session) httpServletRequest
                 .getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
-
-        if (chSession.getSignInInfo().isSignedIn()) {
-
-            final Map<String, Object> sessionData = chSession.getData();
-            final Map<String, Object> signInInfo =
-                    (Map<String, Object>) sessionData.get(SIGN_IN_INFO);
-
-            signInInfo.replace(SessionKeys.SIGNED_IN.getKey(), 1, 0);
-            sessionData.remove(SIGN_IN_INFO);
-
-            final String zxsKey = (String) sessionData.get(".zxs_key");// This is the id of cookie
-                                                                       // stored in redis
-            if (zxsKey != null) {
-                LOGGER.trace("Deleting ZXS info from cache");
-                store.delete(zxsKey);
-            }
-        }
+        oauth2.invalidateSession(chSession, store);
```

Comment:
Okay discussed this with Zain.
I cannot think of any scenario where the store used wouldn't be the default impl because its implicitly used in the filters and session processes, so a custom would also need to overwrite all of that as well.
Additionally, I'd be suspicious of a non reddis store because I don't think other CH apps could communicate with it. And any reasons for changing while maintaining the reddis store such as bug fixing should be done at the default level.
Therefore we should use SessionFactory.getDefaultStoreImpl() rather than requiring the app to provide the store.
This would actually be a better implementation than our bean config because it means its not our responsibility to know what the default store is.

Reply:
It may also be better to pass the request so that we don't rely on libraries that implement this don't have to extract their own session and incase we need additional attributes in the future we don't need to change the signature. 
But haven't discussed that yet.

Reply:
In response to Jon's first comment I agree, however we should implement `SessionFactory.getDefaultStoreImpl()` in our local uk.gov.ch.oauth.session.SessionFactory and have a method that calls off to the `SessionFactory.getDefaultStoreImpl()`. This would allow us to mock when unit testing as the original method is a static one. Furthermore, we could probably do without the store config we have. I would ensure that nothing else uses the config before deleting however.

Reply:
Sooo....we should modify our `oauth.session.SessionFactory ` to contain `SessionFactory.getDefaultStoreImpl()` method?

Reply:
You may need to talk us through what good that actually does wrt the `store`

Reply:
Sure. Set up a call and ping the chat if you need us.